feat: add 'auto' mode to changesNotSentForReview for intelligent retry

MohammedTarigg opened the pull request:
Hey!

I ran into the `Changes cannot be sent for review automatically` error and noticed there's no good way to handle it gracefully. Setting `changesNotSentForReview: true` always works, but then you lose the automatic review submission when it's available.

This PR adds an `auto` option that:
1. Tries to commit normally first
2. If Google blocks it with that specific error, retries with `changesNotSentForReview=true`
3. Logs a warning so you know manual review is needed

Also added a new output `changesNotSentForReviewApplied` so you can trigger Slack notifications or whatever when manual review is required.

Example usage:

```yaml
- name: Upload to Play Store
  id: upload
  uses: r0adkll/upload-google-play@v1
  with:
    changesNotSentForReview: auto
    # ... other inputs

- name: Notify if manual review needed
  if: steps.upload.outputs.changesNotSentForReviewApplied == 'true'
  run: echo 'Manual review required in Play Console'
```

Existing behavior is unchanged - `true`/`false` still work exactly as before.

Related to #84

MohammedTarigg committed:
feat: add 'auto' mode to changesNotSentForReview for intelligent retry

boswelja commented on src/edits.ts:
```diff
@@ -74,6 +75,50 @@ export async function runUpload(
+ * @see https://developers.google.com/android-publisher/api-ref/rest/v3/edits/commit
+ */
+function isChangesNotSentForReviewRequired(error: unknown): boolean {
+    const gaxiosError = error as GaxiosErrorLike;
```
Is `error` not a `GaxiosErrorLike`?

```suggestion
function isChangesNotSentForReviewRequired(error: GaxiosErrorLike): boolean {
```

boswelja commented on action.yml:
```diff
@@ -48,8 +48,8 @@ inputs:
   changesNotSentForReview:
-    description: "Indicates that the changes in this edit will not be reviewed until they are explicitly sent for review from the Google Play Console"
-    default: 'false'
+    description: "Indicates that the changes in this edit will not be reviewed until they are explicitly sent for review from the Google Play Console. Set to 'auto' to automatically retry with this flag if Google Play requires it. Defaults to 'false'"
```
I don't think we need to include "Defaults to x", since we have the default explicitly captured below

boswelja commented on src/edits.ts:
```diff
@@ -108,28 +153,113 @@ async function uploadToPlayStore(options: EditOptions, releaseFiles: string[]):
-            return res.data.id
+                // This is the error we can handle - retry with flag
+                core.warning(
+                    "⚠️  Google Play blocked automatic 'Send for review'. " +
```
Please avoid emotes in console output

boswelja replied:
It looks like there's still emotes and extra whitespace in the console output here

boswelja reviewed (commented):
Thanks for taking the time to open a PR! I've left some comments 😃

MohammedTarigg committed:
Update src/edits.ts

Co-authored-by: Jack Boswell <boswelja@outlook.com>

MohammedTarigg committed:
fix: revert unnecessary quote changes and remove redundant description text

- Revert name, author, icon, color, and default values to single quotes
- Remove 'Defaults to false' from changesNotSentForReview description
- Maintain consistency with original file style

Co-authored-by: Jack Boswell <boswelja@outlook.com>

MohammedTarigg committed:
fix: use error parameter directly in isChangesNotSentForReviewRequired

Co-authored-by: Jack Boswell <boswelja@outlook.com>

boswelja commented:
There hasn't been any activity here for a while, I'm closing this one off. Please resolve conflicts and re-open if you want to get this merged!
